pkg/config/config.go

```diff
@@ -408,28 +408,47 @@ func initConfig(config Config) {
 	config.SetKnown("proxy.no_proxy")
 
 	// Process
-	config.SetKnown("process_config.dd_agent_env")
-	config.SetKnown("process_config.enabled")
-	config.SetKnown("process_config.intervals.process_realtime")
-	config.SetKnown("process_config.queue_size")
-	config.SetKnown("process_config.max_per_message")
-	config.SetKnown("process_config.intervals.process")
-	config.SetKnown("process_config.blacklist_patterns")
-	config.SetKnown("process_config.intervals.container")
-	config.SetKnown("process_config.intervals.container_realtime")
-	config.SetKnown("process_config.dd_agent_bin")
-	config.SetKnown("process_config.custom_sensitive_words")
-	config.SetKnown("process_config.scrub_args")
-	config.SetKnown("process.strip_proc_arguments")
-	config.SetKnown("process_config.windows.args_refresh_interval")
-	config.SetKnown("process_config.windows.add_new_args")
-	config.SetKnown("process.additional_endpoints.*")
-	config.SetKnown("process.container_source")
-	config.SetKnown("process.intervals.connections")
-	config.SetKnown("system_probe_config.enabled")
-	config.SetKnown("system_probe_config.log_file")
-	config.SetKnown("system_probe_config.debug_port")
-	config.SetKnown("system_probe_config.bpf_debug")
+	for _, field := range []string{
+		// Process agent
+		"process_config.dd_agent_env",
+		"process_config.enabled",
+		"process_config.intervals.process_realtime",
+		"process_config.queue_size",
+		"process_config.max_per_message",
+		"process_config.intervals.process",
+		"process_config.blacklist_patterns",
+		"process_config.intervals.container",
+		"process_config.intervals.container_realtime",
+		"process_config.dd_agent_bin",
+		"process_config.custom_sensitive_words",
+		"process_config.scrub_args",
+		"process_config.strip_proc_arguments",
+		"process_config.windows.args_refresh_interval",
+		"process_config.windows.add_new_args",
+		"process_config.additional_endpoints.*",
+		"process_config.container_source",
+		"process_config.intervals.connections",
+		// System probe
+		"system_probe_config.enabled",
+		"system_probe_config.log_file",
+		"system_probe_config.debug_port",
+		"system_probe_config.bpf_debug",
+		"system_probe_config.disable_tcp",
```

Comment:
Nit: Not a huge fan of mixing different way of doing the same thing in the same place as it makes things less readable IMO. If we all agree this is a more readable way of defining the list of configuration option we should align everything at the same time (including `BindEnvAndSetDefault` & friends)

Reply:
👍  I will revert back to the old way for now